tree/ntuple/v7/inc/ROOT/RMiniFile.hxx

```diff
@@ -225,6 +225,10 @@ public:
    std::uint64_t WriteNTupleFooter(const void *data, size_t nbytes, size_t lenFooter);
    /// Writes a new record as an RBlob key into the file
    std::uint64_t WriteBlob(const void *data, size_t nbytes, size_t len);
+   /// Reserves a new record as an RBlob key into the file
+   std::uint64_t ReserveBlob(size_t nbytes, size_t len);
+   /// Write into a reversed record
```

Comment:
Going back into the code, I (re)discovered that the caller actually has to make additional guarantees right now: Because of how file positioning works at the moment in `RFileSimple::Write`, the caller *must* fill the reserved byte range to the end, or the next key is not written where the file format expects it to be. This is currently assured in the limited usage of this feature, and I added corresponding documentation to `ReserveBlob`.

I plan to eventually lift this limitation and have offset-writes work the same as `pwrite` (ie not modify the logical file positioning) - is this ok to leave in the current state for now or do you want me to prioritize this refactoring to get a saner interface from the beginning?

Reply:
Okay, never mind, it actually wasn't that much work (~20 lines), so I included a commit in this PR.